src/frontend/screens/ProjectInviteModal/index.tsx

```diff
@@ -0,0 +1,273 @@
+import * as React from "react";
+import { Alert, StyleSheet, View } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { FormattedMessage, defineMessages, useIntl } from "react-intl";
+import { BottomSheetModal } from "@gorhom/bottom-sheet";
+
+import { MAPEO_BLUE, WHITE } from "../../lib/styles";
+import BottomSheet from "../../sharedComponents/BottomSheet";
+import Text from "../../sharedComponents/Text";
+import { DoneIcon } from "../../sharedComponents/icons";
+import Circle from "../../sharedComponents/icons/Circle";
+import Loading from "../../sharedComponents/Loading";
+import ModalActionButtons from "./ModalActionButtons";
+
+interface ProjectInviteDetails {
+  project: {
+    id: string;
+    name: string;
+  };
+  role: string;
+}
+
+type LoadingStatus = {
+  type: "loading";
+};
+
+type ErrorStatus = {
+  type: "error";
+  info: { error: Error };
+};
+
+type SuccessStatus = {
+  type: "success";
+  info: {
+    inviteDetails: ProjectInviteDetails;
+  };
+};
```

Comment:
I really like this pattern of using UseState() instead of the typical useReducer(). Much easier to read and understand. I will probably also start doing this.

Reply:
yeah disjoint unions are super powerful when your state updates don't involve a lot of logic (in that case useReducer is usually the better solution)